src/core/cache.jl

```diff
@@ -13,12 +13,23 @@ Tracks the last time status of a device changed in a simulation
 mutable struct TimeStatusChange <: AbstractCache
     device_type::Type{<:PSY.Device}
     value::JuMP.Containers.DenseAxisArray{Dict{Symbol, Float64}}
+    units::Dates.TimePeriod
     ref::UpdateRef
+
+    function TimeStatusChange(device_type, value, units, ref)
```

Comment:
add the types to the inputs and make the defauls 

```suggestion
    function TimeStatusChange(device_type::Type{<:PSY.Device}, value::JuMP.Containers.DenseAxisArray{Dict{Symbol, Float64}}, ref::UpdateRef, units::Dates.TimePeriod = Dates.Hour(1))
```